### DISCUSSION TITLE
HTML Output Formatter

### DISCUSSION THREAD

**Discussion by matheusab** [rectorphp/rector#7938] [Ideas]
Hello! First of all, thanks for the excellent work with this library! Definitely a great addition to my toolbox.

I was looking for a way to generate an HTML report like Codeception's code coverage report and, as of #1455, the closest I could get was through the Json Output Formatter.

A couple of days ago I had some time to spare, so I made it possible to output an HTML file like so:

```
vendor/bin/rector process --dry-run src --xdebug --output-format=html
```

The generated report so far contains a summary and side-by-side diffs. Example screenshots:

![image](https://github.com/rectorphp/rector/assets/3750530/24474cb1-aac7-4a61-8dac-46486010e515)

![image](https://github.com/rectorphp/rector/assets/3750530/b4308d47-7ae4-4be8-8547-7bc0427842f2)

> external libs included via CDN: Bootstrap 5 and diff2html

------

So, I have 2 questions:

1: Do you think it would be useful to include this feature in the core library? If so, I can submit a PR
2: If not, I'd publish it as an extension, but first I'd like to know whether this is the proper way of letting Rector be aware of this class: I'm currently doing it through Symfony's DI mechanism like this:
```php
//rector.php 

...

return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->services()->set(HtmlOutputFormatter::class)->autowire()->autoconfigure();
```

Thanks for your attention!
(also, suggestions are welcome!)

**Comment by samsonasik** [comment]
Per @TomasVotruba comment at https://github.com/rectorphp/rector-src/pull/3514#issuecomment-1511308964 , I think this should be on rector extension instead of core.

Yes, You can register service via `$rectorConfig->services();` ,if you want to load a directory, you can use `load()` usage, see https://github.com/rectorphp/rector-src/blob/02b93319906ffa9842cbab8f80d58d8b5576cf43/config/config.php#L76-L82

**Reply by matheusab** [reply]
Right! Thanks for replying ;)
